dependencyInsight is kinda annoying

## What happened?

I'm tinkering with dependencies and trying to run dependencyInsight, but it keeps telling me lockfiles are out of date:

```
./gradlew  :dependencyInsight --configuration unifiedClasspath --dependency log4j
> Task :dependencyInsight FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dependencyInsight'.
> Locked dependencies missing from the resolution result: [org.apache.logging.log4j:log4j-api, org.apache.logging.log4j:log4j-core]. . Please run './gradlew --write-locks'.


```

This forces me to do another `./gradlew --write-locks` and then come back and try again, which is slow.

cc @amandachow 

## What did you want to happen?

I think gradle-consistent-versions could actually just not verify the lockfile integrity if the dependencyInsight task is running (I think gradle already has this special-casing with failOnVersionConflict??)

--- a/src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java
+++ b/src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java
@@ -186,7 +186,14 @@ public class VersionsLockPlugin implements Plugin<Project> {
             }
 
             // Ensure that we throw if there are dependencies that are not present in the lock state.
+            // Unless... dependencies / dependencyInsight was run on the root project
+            ImmutableList<String> tasks = ImmutableList.of(":dependencyInsight", ":dependencies");
             unifiedClasspath.getIncoming().afterResolve(r -> {
+                if (tasks.stream().anyMatch(project.getGradle().getTaskGraph()::hasTask)) {
+                    log.lifecycle("Not checking validity of locks since we are running tasks that inspect "
+                            + "dependencies");
+                    return;
+                }
                 failIfAnyDependenciesUnresolved(r);
 
                 LockState currentLockState = LockStates.toLockState(fullLockStateSupplier.get());

--- a/src/test/groovy/com/palantir/gradle/versions/VersionsLockPluginIntegrationSpec.groovy
+++ b/src/test/groovy/com/palantir/gradle/versions/VersionsLockPluginIntegrationSpec.groovy
@@ -310,6 +310,29 @@ class VersionsLockPluginIntegrationSpec extends IntegrationSpec {
         runTasks(':resolveConfigurations', '--write-locks')
     }
 
+    def 'does not fail if unifiedClasspath is unresolvable but we are running dependencies'() {
+        def notCheckingLocksMessage = "Not checking validity of locks"
+
+        file('versions.lock') << """\
+            org.slf4j:slf4j-api:1.7.11 (0 constraints: 0000000)
+        """.stripIndent()
+
+        addSubproject('foo', '''
+            apply plugin: 'java'
+            dependencies {
+                compile 'org.slf4j:slf4j-api:1.7.20'
+            }
+        '''.stripIndent())
+
+        expect:
+        def result = runTasks('dependencies', '--configuration', 'unifiedClasspath')
+        result.output.contains(notCheckingLocksMessage)
+
+        // Fails if we don't run dependencies
+        def failure = runTasksAndFail(':resolveConfigurations')
+        !failure.output.contains(notCheckingLocksMessage)
+    }
+
     def 'fails if dependency was removed but still in the lock file'() {
         def expectedError = "Locked dependencies missing from the resolution result"
         DependencyGraph dependencyGraph = new DependencyGraph("org:a:1.0", "org:b:1.0")

